### PR TITLE
ICC fixes

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

File: boost/network/protocol/http/client/facade.hpp [hunk 1]
```diff
@@ -122,7 +122,7 @@ class basic_client_facade {
     } else {
       if (boost::empty(content_type_headers)) {
         typedef typename char_<Tag>::type char_type;
-        static char_type content_type[] = "x-application/octet-stream";
+        static char_type* content_type = "x-application/octet-stream";
```
Comment (deanberris) [inline boost/network/protocol/http/client/facade.hpp]: Why do you need to make this change?

File: boost/network/protocol/http/client/facade.hpp [hunk 2]
```diff
@@ -227,7 +227,7 @@ class basic_client_facade {
...
```
Comment (deanberris) [inline boost/network/protocol/http/client/facade.hpp]: Same here.

File: boost/network/protocol/http/client/facade.hpp (hunk 1 above)
Comment (eile) [inline boost/network/protocol/http/client/facade.hpp]: It's the same as #738

**Comment by eile** [comment]
This is #734 and #738 for master

**Comment by tribal-tec** [comment]
#734 is already merged into master (was #729 on 0.13).

**Comment by eile** [comment]
Corrected to only ICC fixes.

**Comment by anonimal** [comment]
@deanberris @glynos status?

**Comment by deanberris** [comment]
LGTM

I'm not sure this is actually a valid change to make -- I don't understand why ICC isn't able to just initialise the character array with a literal, as that's valid C++. However, if this is something that's actively blocking adoption, then I suppose merging it is fine.

**Pull request merged**